lib/news_scraper/trainer/preset.rb

```diff
@@ -0,0 +1,76 @@
+module NewsScraper
+  module Trainer
+    class Preset
+      PROVIDER_PHRASE = 'I will provide a pattern using'
+
+      def initialize(uri:, scrape_patterns:, payload:, data_type_presets:, data_type:)
+        @uri = uri
+        @scrape_patterns = scrape_patterns
+        @payload = payload
+        @data_type_presets = data_type_presets
+        @data_type = data_type
+      end
+
+      def select
+        selected_option = CLI.prompt_with_options(
+          "Select which preset to use for #{@data_type}:",
+          options.keys
+        )
+
+        if selected_option.start_with?(PROVIDER_PHRASE)
+          pattern_type = options[selected_option]
+          return {
+            'method' => pattern_type,
+            'pattern' => CLI.get_input("Provide the #{pattern_type} pattern:")
+          }
+        end
+        return if selected_option == 'skip'
+
+        selected_index = options[selected_option]
+        selected_preset_code = transform_results[selected_index].first
+        @data_type_presets[selected_preset_code]
+      end
+
+      private
+
+      def options
```

Comment:
Nit: I feel like this method name is too vague.

I get that this is all under the `Preset` class, but I feel like calling it `pattern_options` or `preset_options` would be more descriptive.